@rtodling it's the `unit test` that you should be able to see the CDASH output.. but it's just coding norms:

```
./saber2ioda.py:126:1: E302 expected 2 blank lines, found 1
./saber2ioda.py:181:1: E302 expected 2 blank lines, found 0
./saber2ioda.py:216:1: E302 expected 2 blank lines, found 0
./saber2ioda.py:264:1: E305 expected 2 blank lines after class or function definition, found 1
./saber2ioda.py:288:1: W391 blank line at end of file
checking ./saber_download.py
./saber_download.py:1:22: W291 trailing whitespace
./saber_download.py:2:1: W293 blank line contains whitespace
./saber_download.py:9:1: W293 blank line contains whitespace
./saber_download.py:12:1: W293 blank line contains whitespace
./saber_download.py:13:1: E302 expected 2 blank lines, found 1
./saber_download.py:18:1: W293 blank line contains whitespace
./saber_download.py:20:9: W291 trailing whitespace
./saber_download.py:22:36: W291 trailing whitespace
./saber_download.py:24:47: W291 trailing whitespace
./saber_download.py:55:1: E302 expected 2 blank lines, found 1
./saber_download.py:80:1: W391 blank line at end of file
```


note you can run this validation by hand to fix these.  For instance here's a check on `buoy_bufr2ioda.py`:
```
cd $JEDI_SRC/iodaconv/src/conventional
pycodestyle --config=../../.pycodestyle buoy_bufr2ioda.py
```

so that utility `pycodestyle` should be available when the JEDI stack is loaded.

--- a/src/conventional/saber2ioda.py
+++ b/src/conventional/saber2ioda.py
@@ -236,6 +236,7 @@ def main(args):
 
     # Map missing values
     obs_data = map_missing_values(obs_data)
+
     # Time window screening
     if args.date:
         window_start = args.date - timedelta(hours=args.window_offset)

--- a/src/conventional/saber2ioda.py
+++ b/src/conventional/saber2ioda.py
@@ -29,7 +29,6 @@ from pyiodaconv.def_jedi_utils import (
     record_time
 )
 
-
 # Globals
 META_DATA_NAME = iconv.MetaDataName()
 OBS_VAL_NAME = iconv.OvalName()
@@ -264,6 +263,7 @@ def main(args):
         raise ValueError('No valid observations remained.')
 
     logging.debug("Processing complete. Time elapsed: %f", record_time() - tic)
+
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(

--- a/src/conventional/saber2ioda.py
+++ b/src/conventional/saber2ioda.py
@@ -54,7 +54,6 @@ VAR_DIMS = {
     'airTemperature': ['Location'],
 }
 
-
 # Variable attributes
 VAR_ATTRS = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
 VAR_ATTRS['airTemperature', OBS_VAL_NAME]['coordinates'] = 'longitude latitude'
@@ -123,6 +122,7 @@ def time_window_screening(full_obs_data, window_start, window_end):
             filtered_obs_data[key] = numpy.array(val)[valid_time_mask]
     return filtered_obs_data
 
+
 def process_file(file_name):
     if not os.path.isfile(file_name):
         raise FileNotFoundError(f"Input file not found (-i option): {file_name}")
@@ -178,6 +178,8 @@ def get_epoch_time(date_var, time_var):
             epoch_time[i, :] = base_epoch_sec + (times // 1000)
 
     return epoch_time.flatten()
+
+
 def get_data_from_file(obs_file_handle):
     obs_data = {
         ('latitude', META_DATA_NAME): read_variable(obs_file_handle, 'tplatitude', dtype=numpy.float32),
@@ -213,6 +215,8 @@ def get_data_from_file(obs_file_handle):
     logging.debug("Extracted %d observations from file.", nlocs)
 
     return obs_data
+
+
 def main(args):
     if args.verbose:
         logging.basicConfig(level=logging.INFO)
@@ -285,4 +289,3 @@ if __name__ == "__main__":
                           help='Enable verbose messages (INFO level)')
 
     main(parser.parse_args())
-

--- a/src/conventional/saber_download.py
+++ b/src/conventional/saber_download.py
@@ -1,4 +1,5 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python
+
 import os
 import re
 import requests
@@ -6,22 +7,22 @@ import argparse
 from urllib.parse import urljoin
 from datetime import datetime, timedelta
 
-
 BASE_URL = "https://data.gats-inc.com/saber/Version2_0/Level2A"
 SAVE_DIR = "./saber_files"
-            
+
+
 def download_nc_files_by_date(date: datetime):
     year = date.year
     jday = date.timetuple().tm_yday
     jday_str = f"{jday:03d}"
     url = f"{BASE_URL}/{year}/{jday_str}/"
-                
+
     print(f"Accessing: {url}")
-    try:        
+    try:
         response = requests.get(url)
-        response.raise_for_status() 
+        response.raise_for_status()
     except requests.RequestException as e:
-        print(f"Failed to access {url} - {e}") 
+        print(f"Failed to access {url} - {e}")
         return
 
     # Extract all .nc file links from HTML
@@ -78,4 +79,3 @@ def main():
 
 if __name__ == "__main__":
     main()
-

--- a/src/conventional/saber_download.py
+++ b/src/conventional/saber_download.py
@@ -1,12 +1,12 @@
 #!/usr/bin/env python 
-        
 import os
 import re
 import requests
 import argparse
 from urllib.parse import urljoin
 from datetime import datetime, timedelta
-        
+
+
 BASE_URL = "https://data.gats-inc.com/saber/Version2_0/Level2A"
 SAVE_DIR = "./saber_files"
             
@@ -51,6 +51,7 @@ def download_nc_files_by_date(date: datetime):
             print(f"Already exists: {fname}")
 
     print(f"Downloaded {downloaded} files from {url}")
+
 
 def main():
     parser = argparse.ArgumentParser(description="Download SABER Level2A .nc files for a date range.")

